src/boot/efi/stub.c

```diff
@@ -565,13 +565,10 @@ static void refresh_random_seed(EFI_LOADED_IMAGE_PROTOCOL *loaded_image) {
         if (!loaded_image->DeviceHandle)
                 return;
 
+        /* Don't measure again, if sd-boot already initialized the random seed */
         uint64_t loader_features = 0;
         err = efivar_get_uint64_le(MAKE_GUID_PTR(LOADER), u"LoaderFeatures", &loader_features);
-        if (err != EFI_SUCCESS)
-                return;
-
-        /* Don't measure again, if sd-boot already initialized the random seed */
-        if (!FLAGS_SET(loader_features, EFI_LOADER_FEATURE_RANDOM_SEED))
+        if (err == EFI_SUCCESS && FLAGS_SET(loader_features, EFI_LOADER_FEATURE_RANDOM_SEED))
```

Comment:
the conditionalization on `err == EFI_SUCCESS` is redundant, just drop that. we initialize loader_features to zero anyway, hence if reading the efi var fails we'll se a zero flags field, which is all we need.

Reply:
Fixed both, thanks for the review!